posts/nyc-subway-data-with-kafka-streams-and-jhipster-part-2.md

```diff
@@ -0,0 +1,94 @@
+---
+authors:
+- Theo Lebrun
+tags:
+- Apache Kafka
+- JHipster
+- Grafana
+- InfluxDB
+date: 2018-11-07T12:00:00.000Z
+title: "NYC subway data with Kafka Streams and JHipster (part 2 of 2)"
+image: https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2018/11/mta-kafka-logo.png
+---
+
+In the [part 1](https://blog.ippon.tech/nyc-subway-data-with-kafka-streams-and-jhipster-part-1/) of this blog post I explained how to retrieve data from the MTA feed and publish them to Kafka. I will now explain how to easily visualize those data using [InfluxDB](https://www.influxdata.com/) and [Grafana](https://grafana.com/). InfluxDB is an open-source time series database that can be used as a data source for Grafana. Grafana is an open-source, general purpose dashboard and graph composer, which runs as a web application. Grafana will be used to visualize the number of active subways for each lines through a day and see how this number changes with time.
+
+All the code of this blog post can be found on this [GitHub repository](https://github.com/Falydoor/mta-kafka) and I recommend cloning it instead of copying the code examples.
+
+# Saving measurements to InfluxDB
+
+## InfluxDB setup with the microservice
+
+Starting an InfluxDB instance can be achieve using the below Docker command:
+
+```docker run -p 8086:8086 -v $PWD:/var/lib/influxdb influxdb```
+
+A database is required to save the measurements, here is the cURL command to create a database named `mta`:
+
+```curl -G http://localhost:8086/query --data-urlencode "q=CREATE DATABASE mta"```
+
+The microservice can now connect to the instance and starts saving measurements in the `mta` database using the Java driver. In order to use the Java driver, the below Maven dependency must be added first:
+
+```xml
+<dependency>
+    <groupId>org.influxdb</groupId>
+    <artifactId>influxdb-java</artifactId>
+    <version>2.14</version>
+</dependency>
+```
+
+Configuring the java client is very simple and the interface `InfluxDB` will be later used to query the database.
+
+```java
+// Init influxDB
+InfluxDB influxDB = InfluxDBFactory.connect("http://127.0.0.1:8086", "root", "root");
+influxDB.setDatabase("mta");
+```
+
+## Measurements saving
+
+In the [part 1](https://blog.ippon.tech/nyc-subway-data-with-kafka-streams-and-jhipster-part-1/), the Kafka topic `mta-stream` was used to store the results from the streaming process. This topic will be used to save each messages in InfluxDB since a message represents the number of active subways for a given line and time.
```

Comment:
```suggestion
In [part 1](https://blog.ippon.tech/nyc-subway-data-with-kafka-streams-and-jhipster-part-1/), the Kafka `mta-stream` topic was used to store the results from the streaming process. This topic will be used to save each message in InfluxDB since a message represents the number of active subways for a given line and time.
```